system/security/plugins/jwtSecurity/jwtSecurity.cpp

```diff
@@ -1044,9 +1053,9 @@ class CJwtSecurityManager : implements IDaliLdapConnection, public CBaseSecurity
                 }
             }
 
-            if (key && stricmp(key,"Scope") == 0)
+            if (key && strisame(key,"Scope"))
```

Comment:
minor: don't need to check for key=null with strisame
(and on line 1053)

Reply:
verifyToken should be called only under three conditions on a per-user basis:

1. During initial login
2. During token refresh
3. Resurrecting a token from Dali, which happens only when copying a token between processes (e.g. esp stores the token and Dali retrieves it)

The count of the second two items is related to token lifetime, and repeated during a user's session.  A token with a 15-minute lifetime means that those steps will be repeated roughly every 15 minutes, per user.  Actual trigger is based on usage rather than a timer, though:  Permission checks include a check for token timeout, which could trigger a refresh.  Actual token lifetime is defined by the JWT endpoint.

Once verified, a token is stored in a cache and not verified again.

Even with hundreds of concurrent users, I don't think this would be burdensome.

So yeah, I'm pushing back slightly on changing this but if you feel that I should make the change to an owned pointer, let me know.

Reply:
Fair enough, it doesn't sound like it will be called enough to impact performance.
Personally, I'd still change it, it should be a minor change that doesn't affect thread safety, and avoids unnecessary copying of the various hash maps that each JWTUserInfo contains.